src/components/feed/NodeDetails/PluginTitle.tsx

```diff
@@ -1,16 +1,28 @@
 import React from 'react'
 import { useDispatch } from 'react-redux'
-import { TextInput, Button, Title, Alert } from '@patternfly/react-core'
+import { TextInput, Button, Title, Alert, Tooltip } from '@patternfly/react-core'
 import { useTypedSelector } from '../../../store/hooks'
 import { AiFillEdit } from 'react-icons/ai'
 import { setPluginTitle } from '../../../store/pluginInstance/actions'
 import { PluginInstance } from '@fnndsc/chrisapi'
+import { AddNodeProps} from "../AddNode/types";
+import "./NodeDetails.scss";
+import { Cookies } from 'react-cookie'
 
 function getDefaultTitle(selectedPlugin?: PluginInstance) {
   return selectedPlugin?.data.title || selectedPlugin?.data.plugin_name
 }
 
-const PluginTitle = () => {
+const PluginTitle: React.FC<AddNodeProps>  = ( {readOnly}) => {
+  const currentFeedPayload = useTypedSelector((state) => state.feed.currentFeed)
+  const cookie = new Cookies()
+  const user2 = cookie.get('username')
+  const { data: feed} = currentFeedPayload;
+  if(feed && feed.data.creator_username===user2){
```

Comment:
duplication of code. See previous comment about prop v.s. state.

If you see that each button needs to decide `readOnly` for themselves, then a better solution would be to move the logic to the parent which contains all the buttons.